base/compiler/abstractinterpretation.jl

```diff
@@ -1045,10 +1045,11 @@ function sp_type_rewrap(@nospecialize(T), linfo::MethodInstance, isreturn::Bool)
         spsig = linfo.def.sig
         if isa(spsig, UnionAll)
             if !isempty(linfo.sparam_vals)
-                env = pointer_from_objref(linfo.sparam_vals) + sizeof(Ptr{Cvoid})
-                T = ccall(:jl_instantiate_type_in_env, Any, (Any, Any, Ptr{Any}), T, spsig, env)
+                sparam_vals = Any[isa(v, Core.VarargMarker) ? TypeVar(:N, Union{}, Any) :
```

Comment:
That's new in this PR to mark an unbound intvalued parameter (We used to use the TypeVar of the unbound Vararg as the sentinel, but we don't have that anymore). Any other non-Type, non-Int sentinel value would do also. I suppose we could change the C interface from `spvals` to `sptypes` as we've done elsewhere. Since `Const` is now a builtin, that should actually work ok.